Doc/library/typing.rst

```diff
@@ -1785,8 +1828,13 @@ These are not used in annotations. They are building blocks for declaring types.
 
       ``Point2D.__required_keys__`` and ``Point2D.__optional_keys__`` return
       :class:`frozenset` objects containing required and non-required keys, respectively.
-      Currently the only way to declare both required and non-required keys in the
-      same ``TypedDict`` is mixed inheritance, declaring a ``TypedDict`` with one value
+
+      Keys marked with ``Required`` will always appear in ``__required_keys__``
+      and keys marked with ``NotRequired`` will always appear in ``__optional_keys__``.
+
+      For backwards compatibility with Python 3.10 and below,
+      it is also possible to use mixed inheritance to declare both required and
```

Comment:
I'm not sure what "mixed inheritance" means, and the example below makes the meaning of your paragraph clear anyway

```suggestion
      it is also possible to use inheritance to declare both required and
```